docs/spec_draft.md

```diff
@@ -1029,6 +1030,52 @@ specification. Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-documented-ops)
 
+## stablehlo.transpose
+
+`stablehlo.transpose(operand, permutation) -> result`
+
+### Semantics
+
+Permutes the dimensions of `operand` tensor using a permutation of its shape
+`permutation` and produces a `result` tensor. The $i^{th}$ dimension of the
+`result` is given by `permutation`$[i]^{th}$ dimension of the `operand`.
+
+### Operands
+
+| Name | Type |
+|-|-|
+| `operand` | tensor of any supported types |
+| `permutation` | array of type `ui64` |
+
+### Results
+
+| Name | Type |
+|-|-|
+| `result` | tensor of any supported types |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same element type.
+  * (C2) `permutation` is a permutation of `operand`'s shape.
+
+### Examples
+
+```mlir
+// %operand: [
+//            [[1,2], [3,4], [5,6]],
```

Comment:
Let's add spaces after comma to be consistent with the rest of the doc here and below.